src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -212,7 +215,10 @@ export default class QueryEditorUI extends Component<Props, State> {
       : undefined;
     this.onChange(newQuery, dateRange);
     this.onSubmit(newQuery, dateRange);
-    this.setState({ isDataSourcesVisible: enhancement?.searchBar?.showDataSourceSelector ?? true });
+    this.setState({ isDataSetsVisible: enhancement?.searchBar?.showDataSetsSelector ?? true });
+    this.setState({
+      isDataSourcesVisible: enhancement?.searchBar?.showDataSourcesSelector ?? true,
+    });
```

Comment:
is it better to combine the setStates into one call? I forgot if react optimizes this

Reply:
you're right that would be the better approach i'll fast follow